index.bs

```diff
@@ -8083,21 +8079,395 @@ objects.
+            |environment|.\[[Ready]] with the {{void}} value.
+        1.  If |environment| fails to become ready, [=Promise/reject=] |environment|.\[[Ready]]
+            with a "{{NetworkError!!exception}}" {{DOMException}}.
+    1.  Return |environment|.
+</div>
+</div>
+
+<div class="example" id="promise-example-addBookmark">
+
+    <code>addBookmark</code> is a function that requests that the user add the current web page as
+    a bookmark. It's drawn from
+    <a href="https://github.com/domenic/promises-unwrapping/issues/85">some iterative design
+    work</a> and illustrates a more real-world scenario of appealing to environmental asynchrony,
+    as well as immediate rejections.
+
+    <pre highlight="webidl">
+        interface I {
+          Promise&lt;void> addBookmark();
+        };
+    </pre>
+
+<div algorithm="addBookmark">
+
+    The <code>addBookmark()</code> method steps are:
+
+    1.  If this method was not invoked as a result of explicit user action, return
+        [=Promise/rejected promise|a promise rejected with=] a "{{SecurityError}}"
+        {{DOMException}}.
+    1.  If the document's mode of operation is standalone, return
+        [=Promise/rejected promise|a promise rejected with=] a "{{NotSupportedError}}"
+        {{DOMException}}.
+    1.  Let |promise| be [=Promise/a new promise=].
+    1.  Let |info| be the result of getting a web application's metadata.
+    1.  Run the following steps [=in parallel=]:
+        1.  Using |info|, and in a manner that is user-agent specific, allow the end user to make a
+            choice as to whether they want to add the bookmark.
+            1.  If the end-user aborts the request to add the bookmark (e.g., they hit escape, or
+                press a "cancel" button), [=Promise/reject=] |promise| with an "{{AbortError}}"
+                {{DOMException}}.
+            1.  Otherwise, [=Promise/resolve=] |promise| with the {{void}} value.
+    1. Return |promise|.
+</div>
+</div>
+
+<div class="example" id="promise-example-batchRequest">
+
+    Several places in [[SERVICE-WORKERS]] use [=get a promise to wait for all=].
+    <code>batchRequest</code> illustrates a simplified version of one of their uses.
+    It takes as input a [=sequence=] of URLs, and returns a promise for a [=sequence=] of
+    {{Response}} objects created by fetching the corresponding URL.
+    If any of the fetches fail, it will return [=Promise/rejected promise|a promise rejected with=]
+    that failure.
+
+    <pre highlight="webidl">
+        interface I {
+          Promise&lt;sequence&lt;Response>> batchRequest(sequence&lt;USVString> urls);
+        };
+    </pre>
+
+<div algorithm="batchRequest">
+
+    The <code>batchRequest(|urls|)</code> method steps are:
+
+    1.  Let |responsePromises| be « ».
+    1.  [=list/For each=] |url| of |urls|:
+        1.  Let |p| be the result of calling {{WindowOrWorkerGlobalScope/fetch()}} with |url|.
```

Comment:
Maybe better "running the steps specified by the fetch() method"

Reply:
Well, the intent is to make it clear what happens when author code overrides `fetch()`.